gateway/webhook/webhook.go

```diff
@@ -247,6 +248,27 @@ func (bt *batchWebhookTransformerT) batchTransformLoop() {
 				continue
 			}
 
+			// begin
+			if parseAllQueryParams {
+				queryParams := req.request.URL.Query()
+				var tempBody map[string]interface{}
+				err = json.Unmarshal(body, &tempBody)
+
+				if err != nil {
+					req.done <- webhookErrorRespT{err: response.GetStatus(response.RequestBodyReadFailed)}
+					continue
+				}
+
+				tempBody["query_parameters"] = queryParams
+				body, err = json.Marshal(tempBody)
```

Comment:
This additional marshaling logic here seems to be having some impact on the throughput definitely not negligible. One option we are thinking of is to send the query parameters as additional data alongside event payload to the transformer instead of incorporating them within the event body.

@lvrach @cisse21 @chandumlg need help around this. Any other suggestions?

Reply:
> marshaling logic here seems to be having some impact on the throughput definitely not negligible
It would be expectable, but do we know by how much ?

> One option we are thinking of is to send the query parameters as additional data alongside event payload
That is a valid option, but can cause a lot of changes down the line. A similar kind of solution would be to change the payload schema (in a backwards compatible way) to have `body` and `query` at the top level.


A hacky solution for perfromance, is instead of Unmarshal to manipulate json in place. sudo-code:
```go
paramsRawJSON, err = json.Marshal(queryParams)

jsonLastCurl  := body.FindLastIndex("}")
body[:jsonLastCurl] + `"query_parameters":` + paramsRawJSON   + body[body[jsonLastCurl:len(body)] 

```

Reply:
Thanks, @lvrach.

> A similar kind of solution would be to change the payload schema (in a backwards compatible way) to have body and query at the top level.
If this is something we can explore, can you help us?

@arajguha please share the old and new (after incorporating the hacky solution) latency reports (probably via a new private channel).

Reply:
on it